app/database/propagation.go

```diff
@@ -1,24 +1,26 @@
-package service
+package database
 
 import (
 	"net/http"
 	"strings"
 	"time"
 
-	"github.com/France-ioi/AlgoreaBackend/app/database"
 	"github.com/France-ioi/AlgoreaBackend/app/logging"
 )
 
-const PropagationEndpointTimeout = 3 * time.Second
+const endpointTimeout = 3 * time.Second
 
-// SchedulePropagation schedules asynchronous propagation of the given types.
+// StartAsyncPropagation schedules asynchronous propagation of the given types.
 // If endpoint is an empty string, it will be done synchronously.
-func SchedulePropagation(store *database.DataStore, endpoint string, types []string) {
+func StartAsyncPropagation(store *DataStore, endpoint string, types []string) {
```

Comment:
I would move this method into app/service. It can be called right from the endpoint handler in case when the async propagation is chosen. The database package is for database-related things only.